components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -678,17 +675,17 @@ void AdsServiceImpl::Initialize() {
       brave_rewards::prefs::kWalletBrave,
       base::BindRepeating(&AdsServiceImpl::OnPrefsChanged,
                           base::Unretained(this)));
-#if BUILDFLAG(ENABLE_BRAVE_NEWS)
-  profile_pref_change_registrar_.Add(
-      brave_news::prefs::kBraveTodayOptedIn,
-      base::BindRepeating(&AdsServiceImpl::OnPrefsChanged,
-                          base::Unretained(this)));
+  if (base::FeatureList::IsEnabled(brave_today::features::kBraveNewsFeature)) {
+    profile_pref_change_registrar_.Add(
```

Comment:
Brave news pref changes should not be conditional as this feature flag could be changed via griffin while the ads service is running

Reply:
done